internal/locate/region_request.go

```diff
@@ -1423,10 +1422,9 @@ func (s *RegionRequestSender) getRPCContext(
 	switch et {
 	case tikvrpc.TiKV:
 		if s.replicaSelector == nil {
-			selector, err := NewReplicaSelector(s.regionCache, regionID, req, opts...)
-			if err != nil {
-				s.rpcError = err
-				return nil, nil
+			selector, err := NewReplicaSelector(s.regionCache, regionID, req, opts...) //nolint:staticcheck // ignore SA4023, never returns a nil interface value
+			if selector == nil || !selector.isValid() || err != nil {                  //nolint:staticcheck // ignore SA4023, never returns a nil interface value
```

Comment:
I didn't understand, why SA4023 is triggered here?

Reply:
If I remove `selector == nil` check here, the `lint staticcheck ` will pass, otherwise will have following lint error:

<img width="1317" alt="image" src="https://github.com/tikv/client-go/assets/26020263/16815a62-6a8a-4ade-8e35-714f42d8ad9b">

Reply:
😥

Reply:
But I don't want to remove `selector == nil` check here, Since I always afraid to call `selector.isValid()` maybe panic without check`selector == nil`.